pkg/code-generator/cmd/main.go

```diff
@@ -337,20 +337,6 @@ func collectDescriptorsFromRoot(root string, customImports, customGogoArgs, skip
 	addDescriptor := func(f DescriptorWithPath) {
 		mutex.Lock()
 		defer mutex.Unlock()
-		// don't add the same proto twice, this avoids the issue where a dependency is imported multiple times
-		// with different import paths
-		for _, existing := range descriptors {
-			existingCopy := proto.Clone(existing.FileDescriptorProto).(*descriptor.FileDescriptorProto)
-			existingCopy.Name = f.Name
-			if proto.Equal(existingCopy, f.FileDescriptorProto) {
-				// if this proto file first came in as an import, but later as a solo-kit project proto,
-				// ensure the original proto gets updated with the correct proto file path
-				if existing.ProtoFilePath == "" {
-					existing.ProtoFilePath = f.ProtoFilePath
-				}
-				return
-			}
```

Comment:
for anyone reading, the problem is that this logic is within the async code. descriptors can be equal if their names aren't the same, thus the first descriptor found will win out.

by moving this logic out of the async code (and after we sort the descriptors using their name), we can get deterministic results